app/src/main/java/org/cru/godtools/service/AppSeeAnalyticService.java

```diff
@@ -0,0 +1,117 @@
+package org.cru.godtools.service;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+
+import com.appsee.Appsee;
+import com.appsee.AppseeListener;
+import com.appsee.AppseeScreenDetectedInfo;
+import com.appsee.AppseeSessionEndedInfo;
+import com.appsee.AppseeSessionEndingInfo;
+import com.appsee.AppseeSessionStartedInfo;
+import com.appsee.AppseeSessionStartingInfo;
+import com.crashlytics.android.Crashlytics;
+
+import org.cru.godtools.analytics.model.AnalyticsActionEvent;
+import org.cru.godtools.analytics.model.AnalyticsScreenEvent;
+import org.cru.godtools.analytics.model.AnalyticsSystem;
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+
+import java.util.Map;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.UiThread;
+
+public class AppSeeAnalyticService implements Application.ActivityLifecycleCallbacks, AppseeListener {
+    @Nullable
+    private static AppSeeAnalyticService sInstance;
+
+    private AppSeeAnalyticService(Application application) {
+        EventBus.getDefault().register(this);
+        application.registerActivityLifecycleCallbacks(this);
+        Appsee.addAppseeListener(this);
+    }
+
+    @NonNull
+    public static synchronized AppSeeAnalyticService start(Application application) {
+        if (sInstance == null) {
+            sInstance = new AppSeeAnalyticService(application);
+        }
+        return sInstance;
+    }
+
+    @UiThread
+    @Subscribe(threadMode = ThreadMode.MAIN)
```

Comment:
any specific reason you are processing this on the main thread?

Reply:
I am still running some test but it looks like I am not getting consistent reports using ASYNC.

Reply:
k, if ASYNC doesn't work try MAIN_ORDERED for the threadMode

Reply:
that would keep it on the main thread, but guarantee execution order of events being processed.